server/consumer.go

```diff
@@ -4003,15 +4154,25 @@ func (o *consumer) processNextMsgRequest(reply string, msg []byte) {
 	wr.b = maxBytes
 	wr.received = time.Now()
 
-	if err := o.waiting.add(wr); err != nil {
-		// If the client has a heartbeat interval set, don't bother responding with a 409,
-		// otherwise we can end up in a hot loop with the client re-requesting instead of
-		// waiting for the missing heartbeats instead and retrying.
-		if hb == 0 {
-			sendErr(409, "Exceeded MaxWaiting")
+	if o.cfg.PriorityPolicy == PriorityPrioritized {
```

Comment:
Nit: could pass `o.cfg.PriorityPolicy` into `o.waiting.add(..)` similar to `o.waiting.popOrPopAndRequeue(o.cfg.PriorityPolicy)`, then the `If the client has a heartbeat interval set, don't bother responding with a 409,` comment would remain in the same place as well.

Reply:
Was  thinking about it too, but wanted to limit the scope of changes and how much behaviours it affects. However - I feel that was bit too defensive and you're right.